File: solutions/easy/1299.Replace_Elements_with_Greatest_Element_on_Right_Side/1299.Replace_Elements_with_Greatest_Element_on_Right_Side.py
from ast import List


class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        if len(arr) <= 1:
            return [-1]
        max_number = -1

        for i in range(len(arr) - 1, -1, -1):
            temp = arr[i]
            arr[i] = max_number
            max_number = max(temp, max_number)

        return arr

File: solutions/easy/1299.Replace_Elements_with_Greatest_Element_on_Right_Side/1299.Replace_Elements_with_Greatest_Element_on_Right_Side.ts
function replaceElements(arr: number[]): number[] {
  if(arr.length <= 1) return [-1]
  let length = arr.length - 1
  let max = -1

  for(let i = length; i >= 0; i--) {
    let temp = arr[i]
    arr[i] = max
    max = Math.max(temp,max)
  }
  
  return arr
};